### DISCUSSION TITLE
Create new Timesheet record using API

### DISCUSSION THREAD

**Discussion by basileguoz** [kimai/kimai#4874] [General] [answered]
I'm trying to create a new timesheet record using the API provided with swagger.
By following the provided request this should be the body
`{
  "begin": "2024-05-23T11:55:34",
  "end": "2024-05-23T12:55:34",
  "project": 1,
  "activity": 1,
  "description": "string",
  "fixedRate": 0,
  "hourlyRate": 0,
  "user": 1,
  "tags": "string",
  "exported": true,
  "billable": true
}`

But i get this error
`{
    "code": 400,
    "message": "Validation Failed",
    "errors": {
        "errors": [
            "This form should not contain extra fields."
        ],
        "children": {
            "begin": {},
            "end": {},
            "project": {},
            "activity": {},
            "description": {},
            "tags": {}
        }
    }
}`

It seems there is no problem with the request body but some other error.
Am i missing something or the documentation is outdated?

Thanks in advance!

**Comment by kevinpapst** [comment]
Your user account probably has no permission to set the rate fields

**Reply by kevinpapst** [reply, marked as answer]
The error message clearly says which fields are available. exported/billable and the rates are not in there.
The Swagger docs show all possible fields, but they are restricted, depending on the user account used for the API call. Like the Web UI does as well, showing different fields based on the user role

**Reply by basileguoz** [reply]
The request body must have only these
`{ "begin": "2024-05-25T11:55:34",
 "end": "2024-05-25T12:55:34",
  "project": 1, 
  "activity": 1,
   "description": "string",
      "tags": ""`
      
      Thanks
